### PR TITLE
Type fixes

### PR DESCRIPTION
**Pull request by fbs**
Opening this early in case someone  wants to review it  before I do the codegen changes / commit message cleanup. 

Still need to fix struct types, those are a bit more involved, might leave those for a later PR.

### CODE REVIEW

File: src/ast/irbuilderbpf.cpp [hunk 1]
```diff
@@ -264,71 +264,83 @@ void IRBuilderBPF::CreateMapUpdateElem(Map &map, AllocaInst *key, Value *val)
...
-  Value *map_ptr = CreateBpfPseudoCall(map);
+  Value *mapid = CreateBpfPseudoCall(map);
+
+  assert(key->getType()->isPointerTy());
+  assert(mapid->getType()->isIntegerTy());
 
   // int map_delete_elem(&map, &key)
   // Return: 0 on success or negative error
   FunctionType *delete_func_type = FunctionType::get(
       getInt64Ty(),
```
Comment (mmisono) [inline src/ast/irbuilderbpf.cpp]: Isn't `getInt32Ty()` appropriate here (and other helper functions returning int)?

**Comment by fbs** [comment]
Blocked on #1108

File: src/ast/irbuilderbpf.cpp (hunk 1 above)
Comment (fbs) [reply src/ast/irbuilderbpf.cpp]: ah good catch. I'm not sure if it actually matters for bpf but   it won't hurt to fix it :)

**Comment by fbs** [comment]
Going to split this up  into a few smaller PRs

**Pull request closed without merging**